tests/compile-fail/cast_fn_ptr1.rs

```diff
@@ -1,9 +1,9 @@
 fn main() {
-    fn f() {}
+    fn f() {} //~ ERROR calling a function with more arguments than it expected
```

Comment:
Interestingly, the span moved here. Might have to do with switching to a `try` block?

Reply:
Ah it probably does, `throw_*!` macros internally use `return` so `try` no longer catches them *oops*